File: FirstUniqueCharacterinaString.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        
        vector<int> m(26, 0);
        
        for(char c : s) m[c - 'a']++;
        
        for(int i = 0; i < s.size(); i++)
        	if(m[s[i] - 'a'] == 1) return i;
        return -1;
        
    }
};

int main() {
	
	string s;
	cin>>s;
	
	Solution *solution = new Solution();
	cout<<solution->firstUniqChar(s);
	
	return 0;
	
}

File: FirstUniqueCharacterinaString.py
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        m = {}
        for ch in s:
            if ch not in m:
                m[ch] = 0
            m[ch] += 1
        for i in range(len(s)):
            if m[s[i]] == 1:
                return i
        return -1


if __name__ == '__main__':
    s = input()
    solution = Solution()
    print(solution.firstUniqChar(s))
